src/unify/mod.rs

```diff
@@ -199,43 +197,31 @@ pub struct Snapshot<S: UnificationStore> {
 
 impl<K: UnifyKey> VarValue<K> {
     fn new_var(key: K, value: K::Value) -> VarValue<K> {
-        VarValue::new(key, value, key, key, 0)
+        VarValue::new(key, value, 0)
     }
 
-    fn new(parent: K, value: K::Value, child: K, sibling: K, rank: u32) -> VarValue<K> {
+    fn new(parent: K, value: K::Value, rank: u32) -> VarValue<K> {
         VarValue {
             parent: parent, // this is a root
             value: value,
-            child: child,
-            sibling: sibling,
             rank: rank,
         }
     }
 
-    fn redirect(&mut self, to: K, sibling: K) {
-        assert_eq!(self.parent, self.sibling); // ...since this used to be a root
+    fn redirect(&mut self, to: K) {
         self.parent = to;
-        self.sibling = sibling;
     }
 
-    fn root(&mut self, rank: u32, child: K, value: K::Value) {
+    fn root(&mut self, rank: u32, value: K::Value) {
         self.rank = rank;
-        self.child = child;
+        // self.child = child;
```

Comment:
Maybe just kill it instead of commenting it out?

Reply:
oops